File: leetcode-cn-go/solution0001.go
func twoSum(nums []int, target int) []int {
    res := make([]int, 2)
    size := len(nums)
    if size < 2 {
        return res
    }
    cache := make(map[int]int)
    for i := 0; i < size; i++ {
        key := target - nums[i]
        if v, ok := cache[key]; ok {
            res[0] = v; 
            res[1] = i;
            return res
        }
        cache[nums[i]] = i
    }
    return res
}

File: leetcode-cn/Solution0001.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];

        if (nums.length < 2) {
            return res;
        }

        // cache
        Map<Integer, Integer> cache = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int key = target - nums[i];
            if (cache.containsKey(key)) {
                return new int[]{cache.get(key), i};
            }
            cache.put(nums[i], i);
        }
        return res;
    }
}
